core-blocks/file/edit.js

```diff
@@ -59,6 +61,10 @@ class FileEdit extends Component {
 				allowedType: '*',
 				filesList: [ file ],
 				onFileChange: ( [ media ] ) => this.onSelectFile( media ),
+				onError: ( message ) => {
+					this.setState( { hasError: true } );
```

Comment:
I had to use local state here because we don't support calling `this.props.setAttributes()` in `componentDidMount`.